### ISSUE TITLE
[Rule Tuning] discovery_virtual_machine_fingerprinting - Recommend auditd settings

### ISSUE THREAD

**Issue by themrb** [elastic/detection-rules#100]
## Description
I suggest adding information into description for rule linux/discovery_virtual_machine_fingerprinting on the Auditbeat / auditd configuration required to properly detect the activity.

I have had difficulty test triggering rule using the default Auditbeat 7.8 configuration, the following command does not trigger the rule, because the process execution is not logged.
`cat /sys/class/dmi/id/bios_version`

**Comment by bm11100** [comment]
Hi @themrb! Thank you for reaching out! Perhaps you could provide us with a little more information on your Audibeat / system configuration. 

- Which version of Auditbeat are you using? Also, what stack version are you trying this on?

- Is your configuration detecting any activity? For example file_integrity activity in the `/etc` path?

- Have you tested other rules? Successfully or unsuccessfully?

- Which operating system are you using? 

Without knowing the answers to those questions, I do have one guess. If you are using the default Auditbeat configuration, it has the auditd modules commented out by default. If yours are commented out, can you please uncomment the ones I have highlighted below and let us know if this helps resolve your issue?

Thanks!

![image](https://user-images.githubusercontent.com/56412096/89326723-c5125300-d658-11ea-8f9b-0f81f73877cd.png)

**Comment by themrb** [comment]
Thanks for the suggestions @bm11100!

My stack / os versions are:
kibana 7.8.0
elasticsearch 7.8.0
auditbeat 7.8.0
centos 7-8.2003.0.el7
audit 2.8.5 (4.el7)

Auditbeat is logging quite a bit of network and command activity. I was able to trigger the detection rule by using 'less' keeping the process open until the log entry came through - if I exited 'less' too quickly, I did not get the log event (or the signal trigger). This suggests to me that Auditbeat is scraping the process list by default.

I did have those auditd modules commented out by default so I just gave that a go - but uncommenting them, restarting Auditbeat, then retriggering 'cat' didn't provide the log event.

**Comment by bm11100** [comment]
Thanks for the info @themrb. 

Are you using version 1 of this rule that shipped with 7.8? Or version 2 (see both below), that is currently the updated version in this repo, that will ship with 7.9?

Version 1 - 

```
event.action:executed and
process.args:("/sys/class/dmi/id/bios_version" or
"/sys/class/dmi/id/product_name" or "/sys/class/dmi/id/chassis_vendor"
or "/proc/scsi/scsi" or "/proc/ide/hd0/model") and not user.name:root
```

Version 2 -

```
event.category:process and event.type:(start or process_started) and
  process.args:("/sys/class/dmi/id/bios_version" or
                "/sys/class/dmi/id/product_name" or
                "/sys/class/dmi/id/chassis_vendor" or
                "/proc/scsi/scsi" or
                "/proc/ide/hd0/model") and
  not user.name:root
```

If you are using version 2, there will be some inconsistencies since it's intended for 7.9, and you are using a 7.8 beat. 

I spoke to my colleague @dcode about this, and if you are using version 1 of the rule, it's likely you have auditd running by default on CentOS, which will interfere with Auditbeat.

See the link below-
https://www.elastic.co/guide/en/beats/auditbeat/current/auditbeat-module-auditd.html

> When running Auditbeat with the auditd module enabled, you might find that other monitoring tools interfere with Auditbeat. For example, you might encounter errors if another process, such as auditd, is registered to receive data from the Linux Audit Framework. 

You'll need to stop the auditd service and disable it from starting on boot if you plan to use Auditbeat with the auditd module. 

See if auditd is running:
`service auditd status`

Stop the auditd service:
`service auditd stop`

Disable auditd from starting on boot:
`chkconfig auditd off`

Once this is complete, go ahead and start Auditbeat.

Give this a go and let us know if it works for you. Thanks!

**Comment by dcode** [comment]
@themrb, I'm curious, are you getting any audit events at all from auditbeat?  You can check in Kibana on the "Discover" app by searching the `auditbeat-*` index for `event.module: auditd` and look for your system.

On CentOS/RedHat 7 systems, I've configured auditbeat before and found I was only getting events from `system` and `file_integrity` modules. The problem is that only one process can connect to the kernel's audit socket and `auditd` runs by default on RedHat/CentOS systems in such a way that you can't directly stop it.

## Troubleshooting the auditd module

Here's how you can troubleshoot this. I created a new CentOS 7 (7.8.2003) VM and installed auditbeat 7.8.1 using the [directions](https://www.elastic.co/guide/en/beats/auditbeat/current/auditbeat-installation.html). I uncommented the default auditd rules so that it matches @bm11100's screenshot above. Then I start auditbeat with `systemctl start auditbeat`.

If you look at the output of `journalctl -u auditbeat` in this scenario, you'll see the following:

```
ERROR  [auditd]  auditd/audit_linux.go:155  Failure receiving audit events   {"error": "failed to set audit PID. An audit process is already running (PID 2647)"}
```

Auditbeat will continue to work for the other modules, but this line indicates that it cannot connect to the kernel's audit socket since another process already owns it.

## Fixing the auditd module

To fix this, you need to disable auditd from starting (you'll get no more events in `/var/log/audit/audit.log`), then stop the auditd daemon, then start auditbeat.

```bash
sudo systemctl disable auditd     # Prevents auditd from starting on next boot
sudo service auditd stop              # systemd won't let you stop this directly
sudo systemctl enable auditbeat
sudo systemctl restart auditbeat
```

Assuming you weren't getting the actual audit, you'll now see _a lot_ more events coming into your cluster. I particularly recommend leaving SELinux enabled and you get lots of fun context that you otherwise wouldn't get, like the original user that logged in, even if they `sudo`'d to root or another user.

I hope that's helpful!

**Comment by themrb** [comment]
Really appreciate the extra effort to help me here - auditd wasn't working in exactly the way you predicted @dcode , and I was also running Version 2 of the rule as @bm11100 identified. I've fixed both now. Should I be running Version 1 of all Linux rules for 7.8?

Would it be worth noting in the rule description that accurate detection relies on setting the auditd module configuration, and running the stack at version 7.9? This would help organisations understand what needs to be in place for the rule to be functioning - I have seen a number of organisations get a false sense of security thinking SIEM rules are protecting them, but they don't have the required event types.

More broadly, I was hoping to use this repo to collaborate and regularly update to the latest security community detections, like an elasticsearch-specific version of the sigma project. Will I need to stay on a future release (i.e. 7.9 right now) for this?

**Comment by dcode** [comment]
@themrb thanks for the feedback. We've discussed within the team how to best share with the community our recommended settings for a given rule. We're definitely refining that aspect continuously. 

As for tracking versions, at this exact moment in time we're still snapshotted with Elastic stack releases. We recognize users such as yourself that wish to track signatures between releases. In general, that should be an acceptable workflow. @bm11100 asked about version especially because we just changed the logic of the rule. We do test thoroughly, but if you found an edge case that we didn't plan for, it's helpful to know which logic you used.

I'm glad the pointers helped your issue. I'll reach out to the auditbeat team to see how we can improve the docs to avoid others from having the same problem.
